include/openthread/instance.h

```diff
@@ -255,6 +255,49 @@ OTAPI void OTCALL otRemoveStateChangeCallback(otInstance *aInstance, otStateChan
                                               void *aCallbackContext);
 
 /**
+ * This enumeration defines the constants used in `otChildTableCallback` to indicate whether a child is added or
+ * removed.
+ *
+ */
+typedef enum otChildTableEvent
+{
+    OT_CHILD_TABLE_EVENT_CHILD_ADDED,        ///< A child is being added.
+    OT_CHILD_TABLE_EVENT_CHILD_REMOVED,      ///< A child is being removed.
+} otChildTableEvent;
+
+/**
+ * This function pointer is called to notify that a child is being added to or removed from child table.
+ *
+ * @param[in]  aEvent      A event flag indicating whether a child is being added or removed.
+ * @param[in]  aChildInfo  A pointer to child information structure.
+ *
+ */
+typedef void (*otChildTableCallback)(otChildTableEvent aEvent, const otChildInfo *aChildInfo);
+
+/**
+ * This function gets the child table callback function.
+ *
+ * @returns  The callback function pointer.
+ *
+ */
+otChildTableCallback otGetChildTableCallback(otInstance *aInstance);
+
+/**
+ * This function sets the child table callback function.
+ *
+ * The provided callback (if non-NULL) will be invoked when a child entry is being added/removed to/from the child
+ * table. Subsequent calls to this method will overwrite the previous callback. Note that this callback in invoked
+ * while the child table is being updated and always before the `otStateChangedCallback`.
+ *
+ * @param[in] aCallback                  A pointer to callback handler function.
+ *
+ * @retval OT_ERROR_NONE                 The callback was updated successfully.
+ * @retval OT_ERROR_DISABLED_FEATURE     The feature is disabled (MTD build).
+ *
+ */
+otError otSetChildTableCallback(otInstance *aInstance, otChildTableCallback aCallback);
```

Comment:
Should these new APIs live in [include/openthread/thread_ftd.h](https://github.com/openthread/openthread/blob/master/include/openthread/thread_ftd.h)?

Reply:
Good idea! They are really part of MLE and only applicable/usable under Thread FTD operation. The only downside I see is that they won't be close to  `otStateChangedCallback` callback definitions. Still I agree  they'd fit better in `thread_ftd.h`. Will move/rename them.